### PR TITLE
Jvm packaging

### PR DESCRIPTION
**Pull request by DariusIMP**
Overview
---

This PR contains the logic to a JVM package to GitHub packages, similarly to what we already have for the Android target.

Implementation
---

As opposed to Android targets, where the Android SDK already packages the binaries inside the package and where the Android operating system dynamically loads the library based on the platform on where the application is running, those mechanisms need to be implemented programmatically by us when dealing with JVM.

The goals of this PR were the following:
* To be able to cross compile Zenoh JNI for multiple targets
* Store the libraries in a compressed form in the JAR package (as to reduce the package size)
* Load the compressed library depending on the platform on top of which the application is running

The crosscompilation is done similar to zenoh's, using Github actions with multiple runners with different operating systems (however `cross` needed to be used for some targets in the workflow to avoid the ring library issue). The built zenoh_jni libraries are then uploaded to be later downloaded and used by a publication job.

So far, these are the targets the JVM publication would support:
- x86_64-unknown-linux-gnu
- aarch64-unknown-linux-gnu
- x86_64-apple-darwin
- aarch64-apple-darwin
- x86_64-pc-windows-msvc

### CODE REVIEW

**Comment by evshary** [comment]
I think it's quite good for me.

File: zenoh-jni/.cargo/config
Comment (Mallets) [inline zenoh-jni/.cargo/config:1]: Why an empty file? Perhaps to be added to `.gitignore`?

File: zenoh-kotlin/build.gradle.kts [hunk 1]
```diff
@@ -82,24 +82,12 @@ kotlin {
             kotlinOptions.jvmTarget = "11"
         }
         testRuns["test"].executionTask.configure {
-            val zenohPaths = "../zenoh-jni/target/release"
+            val zenohPaths = "../zenoh-jni/target/debug"
```
Comment (Mallets) [inline zenoh-kotlin/build.gradle.kts:85]: I think we should be building in release, any reason on why building on debug?

File: zenoh-jni/.cargo/config
Comment (DariusIMP) [reply zenoh-jni/.cargo/config:1]: My bad, it slipped through.

File: zenoh-kotlin/build.gradle.kts (hunk 1 above)
Comment (DariusIMP) [reply zenoh-kotlin/build.gradle.kts:85]: Even for testing? I did it to reduce the execution time of the CI. In Rust, errors may arise in release mode that wouldn't be noticed on debug?

Comment (Mallets) [reply zenoh-kotlin/build.gradle.kts:85]: My bad, I didn't notice that this was the build for testing. `debug` is fine.

**Review by Mallets — APPROVED** [review]

**Pull request merged**